src/main/java/Problem2.java

```diff
@@ -2,6 +2,35 @@
 
     // Do not change signature (function name, parameters)
     public static void bubbleSort(int[] data) {
-        // Add your solution
+        if (data == null || data.length == 0) {
+            return;
+        }
+        // From smallest to largest
+        boolean sorted = false;
+        int temp;
+        while (!sorted) {
+            sorted = true;
+            for (int i = 0; i < data.length - 1; i++) {
+                if (data[i] > data[i + 1]) {
+                    temp = data[i];
+                    data[i] = data[i + 1];
+                    data[i + 1] = temp;
+                    sorted = false;
+                }
+            }
+        }
+         // I copied the code from this website: https://stackabuse.com/bubble-sort-in-java/
```

Comment:
if you are still not clear about how the code works or any other question, please ping me on discord.